packages/block-library/src/navigation-link/edit.js

```diff
@@ -117,6 +117,8 @@ function getSuggestionsQuery( type, kind ) {
 			return { type: 'term', subtype: 'category' };
 		case 'tag':
 			return { type: 'term', subtype: 'post_tag' };
+		case 'format':
+			return { type: 'post-format' };
```

Comment:
For folks wondering about the kebab case:

https://github.com/WordPress/gutenberg/blob/df8e7307404818446c3ffd0d86ac6584fa15a911/packages/editor/src/components/provider/use-block-editor-settings.js#L79

FYI on behavior, it also looks like we removed the labels in search results if all results are the same:
https://github.com/WordPress/gutenberg/issues/24839

So this is expected behavior:

| Custom Link Search | No labels with all post formats |
|-----|-----|
| <img width="515" alt="custom link search mixed types" src="https://user-images.githubusercontent.com/1270189/113185874-d6255b00-920b-11eb-8d9e-32f88a05d36e.png"> | <img width="595" alt="post format same types" src="https://user-images.githubusercontent.com/1270189/113185892-db82a580-920b-11eb-99db-4fde6d78aec8.png"> |